lib/online_order.rb

```diff
@@ -0,0 +1,106 @@
+require_relative '../lib/order'
+
+require 'csv'
+require 'awesome_print'
+
+module Grocery
+  class  OnlineOrder < Grocery::Order
+    attr_reader :customerid, :status
+
+    #overwrites initialize method for Order
+    def initialize(id, products, customerid, status)
+      super(id, products)
+      @customerid = customerid
+      @status = status
+    end
+
+    #returns a list of OnlineOrder instances where the value of the customer's ID matches the passed parameter.
+    def self.find_by_customer(customer_id)
+      online_orders_csv = CSV.read("support/online_orders.csv", 'r',headers: true).to_a
+
+      if customer_id > online_orders_csv.length
+        return nil
+      end
+
+      if customer_id >=1
+        specific_online_order = online_orders_csv [customer_id-1]
+      elsif customer_id < 0
+        specific_online_order = online_orders_csv[customer_id]
+      else
+        return nil
+      end
+      return specific_online_order
+    end
+
+
+    #overwrites all method for Order
+    def self.all
+      online_orders=[]
+
+      online_orders_csv = CSV.read("support/online_orders.csv", 'r',headers: true).to_a
+
+      online_orders_csv.each do |line|
+        online_orders << Grocery::OnlineOrder.new(*line)
+      end
```

Comment:
This has the same issue as `Order.all`: you're not parsing the list of products.